src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition.hs

```diff
@@ -12,11 +12,11 @@ import Swarm.Game.Scenario.Topography.Structure.Recognition.Log
 import Swarm.Game.Scenario.Topography.Structure.Recognition.Registry
 import Swarm.Game.Scenario.Topography.Structure.Recognition.Type
 
-data StructureRecognizer = StructureRecognizer
-  { _automatons :: RecognizerAutomatons
-  , _foundStructures :: FoundRegistry
+data StructureRecognizer b en a = StructureRecognizer
```

Comment:
I think there should be a comment here explaining what the different type parameters represent.